pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -221,29 +221,26 @@ def set_config(self, cfg: SimConfig) -> None:
         kraus_ops = []
         self._collapse_ops = []
         if "dephasing" in self.config.noise:
+            prob = self.config.dephasing_prob
             n = self._size
-            if self.basis_name == "all" and self.dim == 3:  # three-level system
-                prob = self.config.dephasing_prob / 3
-                if prob > 0.1 and n > 1:
-                    warnings.warn(
-                        "The dephasing model is a first-order approximation in the"
-                        f" dephasing probability. p = {3*prob} is too large for "
-                        "realistic results.",
-                        stacklevel=2,
-                    )
-                M_0 = np.sqrt(prob/(1-prob)) * qutip.qeye(3)
-                M_i = [np.sqrt(prob) * qutip.ket(str(i), 3) * qutip.bra(str(i), 3) for i in range(3)]
-                kraus_ops.append(M_0)
-                kraus_ops.append(M_i)
+            if prob > 0.1 and n > 1:
+                warnings.warn(
+                    "The dephasing model is a first-order approximation in the"
+                    f" dephasing probability. p = {prob} is too large for "
+                    "realistic results.",
+                    stacklevel=2,
+                )
+            if (
+                self.basis_name == "all" and self.dim == 3
+            ):  # three-level system
+                k = np.sqrt(prob * (1 - prob) ** (n - 1))
+                m_0 = np.sqrt(1 - prob) * qutip.qeye(3)
```

Comment:
This is already being added to the `_collapse_ops` in line 246, so you shouldn't do it again